client/pages/restaurant-page.jsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import Container from 'react-bootstrap/Container';
+import fetch from 'node-fetch';
+import Spinner from './spinner';
+import Row from 'react-bootstrap/Row';
+import Column from 'react-bootstrap/Col';
+
+export default class Restaurant extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedRestaurant: null
+    };
+  }
+
+  componentDidMount() {
+    fetch('/api/random')
+      .then(response => response.json())
+      .then(data => this.setState({
+        selectedRestaurant: data
+      }));
+  }
+
+  render() {
+    const { selectedRestaurant } = this.state;
+    const categories = [];
+
+    function tConv24(time24) {
+      let ts = time24;
+      const H = +ts.substr(0, 2);
+      let h = (H % 12) || 12;
+      h = (h < 10) ? ('0' + h) : h;
+      const ampm = H < 12 ? ' AM' : ' PM';
+      ts = h + ':' + ts.substr(2, 3) + ampm;
+      return ts;
+    }
+
+    if (!selectedRestaurant) {
+      return (
+      <Spinner/>
+      );
+    }
```

Comment:
You should execute guards like this right away. It's confusing to read lines 26-36 only for it to possibly not matter.